### ISSUE TITLE
Support non Latin script fonts

### ISSUE THREAD

**Issue by Brahmasmi** [google/material-design-lite#1717]
Please add support for non Latin script fonts (Noto Sans), with a mechanism to change the script font based on user selection.

Is this a new component, or a missing feature in an existing one?
This is a missing feature in Typography.

Material Design spec URL for the component (if any):
https://www.google.com/design/spec/style/typography.html. Material design recommends Noto Sans for non Latin script fonts. 

What does this component or feature allow you to do which isn’t possible at the moment?
When built, it will allow serving locales using non Latin scripts. Currently, it is not possible to use non Latin script fonts. The file at src/typography/_typography.scss supports only Latin script fonts (Roboto et al).

Please provide any URLs or screenshots of good examples of usage of this component or feature that you may have come across.
A typical use case would be a global site serving multiple locales. For example, support.google.com with its support for multiple locales via a dropdown selection.

There is some "super early" work (author's own words) at #1435. We would appreciate a robust solution.

**Comment by surma** [comment]
This is an still ongoing effort, as you already found out yourself. I’m sorry we haven’t shown any progress here, but we are working on it and it’s on the roadmap for sure :) 

Closing this in favor of #1435.

**Comment by Brahmasmi** [comment]
Thank you @surma for sharing that. And please don't apologize - we are thankful to the entire team for all your help in this gargantuan effort.

1) Can we change our font stack from Roboto, Helvetica... to Roboto, **Noto**, Helvetica... ?
On https://developer.mozilla.org/en/docs/Web/CSS/font-family, the "Note" suggests that the browser selects the font on a per-glyph basis. So if the glyph is present in the first font in the font-family, the browser uses that font else the font family is further traversed till it finds the requisite glyph. I presume it displays a tofu in the default case.

Now, the material design spec (https://www.google.com/design/spec/style/typography.html) recommends the web font stack to be Roboto, Noto and then sans-serif. This is also implemented in Polymer (https://github.com/PolymerElements/paper-styles/blob/master/typography.html#L25). However, our font stack is 'Roboto', 'Helvetica', 'Arial', sans-serif !default (https://github.com/google/material-design-lite/blob/master/src/_variables.scss#L64-L65).

Let me know in case you want me to open a new issue for this request.

2) Would it be possible for us to tag #1435 onto a milestone?
I am requesting this in order to allow the effort to be tracked. In case this is being tracked internally, please disregard this request.

**Comment by surma** [comment]
> Can we change our font stack from Roboto, Helvetica... to Roboto, Noto, Helvetica... ?

Interesting suggestion. I like it. Not sure about the implications. @addyosmani @sgomes WDYT?

> 2) Would it be possible for us to tag #1435 onto a milestone?
I am requesting this in order to allow the effort to be tracked. In case this is being tracked internally, please disregard this request.

Not sure what you mean by being tracked here. You can follow the issue to get informed about updates.

**Comment by sgomes** [comment]
@surma I don't think there would be any grave implications, but I'm not sure it's just enough to change the font stack. We'd likely have to import the font from Google Fonts as well, to have any hope of coverage across devices.

**Comment by Brahmasmi** [comment]
From my understanding of the "Note" on https://developer.mozilla.org/en/docs/Web/CSS/font-family, the addition of Noto font will cause the following behavior modification:
1) If there was a glyph which was earlier displayed using Helvetica or later fonts, it may now be displayed using Noto.
2) If there was a glyph which was earlier displayed as a tofu, it may now be displayed using Noto.

The first modification is expected. The second modification is where the change shines - it would allow non Latin script fonts to be displayed using Noto instead of being shown as tofu (**no to**fu = Noto).

Should MDL start loading Noto from Google fonts?
Not in this change. I assume that will be addressed by #1435 in the long term.

So how does this change help?
In the short term, the developer can load the appropriate script font from Google Fonts and escape tofu land. For example, if a developer wants to display Sanskrit text, he can load the Devanagari subset of Noto via Google Fonts (<link rel='stylesheet' href='https://fonts.googleapis.com/css?family=Noto+Sans&subset=devanagari,latin'>) and the browser will display non-tofus.

In summary, the new stack provides the developer an opportunity to display non Latin script fonts. In the old stack, the developer is destined to the all-welcoming all-encompassing tofu land, even when the appropriate Noto font is web-loaded.

> Not sure what you mean by being tracked here

I presume that if and when the issue is put on a milestone, its chances of witnessing progress increase by the virtue of it being part of the milestone. It may not make the milestone and may get re-tagged to future releases, but it will be tracked by the project (unless it goes WontFix). In either case, no sweat.

**Comment by Garbee** [comment]
As far as issue tracking goes, everything is being "Tracked" if the issue remains open. We are simply prioritizing work on milestone assigned issues to try and hit the fixes for the given milestone. We don't tag to a milestone just to later put it off to another one. That is only done when necessary and the issue simply can't be resolved in a reasonable manner before the milestone lands.
